api/envoy/api/v2/endpoint/endpoint_components.proto

```diff
@@ -29,6 +29,10 @@ message Endpoint {
     // check port. Setting this with a non-zero value allows an upstream host
     // to have different health check address port.
     uint32 port_value = 1 [(validate.rules).uint32 = {lte: 65535}];
+
+    // If this is set to true, and the hostname is not empty, then the hostname in this endpoint
+    // will be used as the host header when doing health checks.
```

Comment:
Is this regardless of the setting of auto host rewrite? I can see the use case, but what if someone wants an alternative host in health check and wants regular auto host rewrite behavior, I don't think that's now possible.

Reply:
With this change one can leave this false, set the hostname on the endpoint, and use the hostname in the health check config in the cluster. This allows custom hostname for health checks but on the cluster level.

Alternatively this can be a `string hostname` as well instead of the book.

No strong preference on my behalf - open to change if you think string is better